foreman/foreman.spec

```diff
@@ -621,6 +641,10 @@ if [ ! -e %{_datadir}/%{name}/config/initializers/encryption_key.rb -a \
   ln -s %{_sysconfdir}/%{name}/encryption_key.rb %{_datadir}/%{name}/config/initializers/
 fi
 
+# copy pregenerated apipie cache files
```

Comment:
I feared you'd say it was that way around :)  Yeah, we need to sort this in core or apipie-rails then, rather than packaging, as we need to be able to handle core-without-plugin updates reliably.  I've no idea about routes, probably depends on how the plugin is written.

One idea before we go the copying-in-rake route.. today, we serve up precompiled assets from multiple plugin directories: https://github.com/theforeman/foreman/blob/1.7.1/config/environments/production.rb#L140-L147

Would it be possible to serve up static apipie files in the same way - if we can order the plugins above core?  Then we could build the index of all the plugins, and even avoid copying them out of the plugin package.

If not, then copying in the rake task that builds the index is the next best thing to do.  We would need to support a directory location outside of the gem installation root due to support Debian packaging (same reason as https://github.com/theforeman/foreman/commit/d43379f323c507d19a2e8f051cd2e767be84125e), i.e. stick with /usr/share/foreman/public/apipie-cache/plugin/%{plugin-name}.